lib/autoupdate/package_url.go

```diff
@@ -56,7 +57,10 @@ const (
 
 // MakeURL constructs the package download URL from template, base URL and revision.
 func MakeURL(uriTmpl string, baseURL string, pkg string, version string, flags InstallFlags) (string, error) {
-	tmpl, err := template.New("uri").Parse(uriTmpl)
+	// We add sprig functions to have access to `semver` and be able to change the template in tests based on
+	// the version requested (e.g. pull pre-releases version from the dev repo, and official versions
+	// from the prod one).
+	tmpl, err := template.New("uri").Funcs(sprig.HermeticTxtFuncMap()).Parse(uriTmpl)
```

Comment:
We're removing the external template override in https://github.com/gravitational/teleport/pull/51383, will that be a problem for this testing use case?

Reply:
For now we can add only `semver` and `semverCompare` to solve this use-case, but we'll have to document this somewhere. I can vendor the semver code.